src/app/blocks/audio-player-block/audio-player-block.component.html

```diff
@@ -20,12 +20,12 @@
                        [models]="[model]"
                        [context]="context"></app-blocks-workflow>
 </div>
-<div *ngIf="!hasError && onPlayBlocks.length > 0">
+<div *ngIf="!hasError && onPlayBlocks.length > 0 && isPlaying">
   <app-blocks-workflow [blocks]="onPlayBlocks"
                        [models]="[model]"
                        [context]="context"></app-blocks-workflow>
 </div>
-<div *ngIf="!hasError && onPauseBlocks.length > 0">
+<div *ngIf="isPlaying == false && onPauseBlocks.length > 0">
```

Comment:
I think pause events need to happen even if there is an error, if a user cannot play something e.g: due to network error, they might not be happy to keep paying, so those events need to keep flowing.

Reply:
I think the onPauseBlock's should only exist if isPlaying returns false, hope you agree this is a sensible pattern for doing it.

Reply:
agreed! nice one